react/components/Slide.js

```diff
@@ -0,0 +1,130 @@
+import React, { Component } from 'react'
+import EventListener from 'react-event-listener'
+import classnames from 'classnames'
+import PropTypes from 'prop-types'
+import debounce from 'debounce'
+import styles from './styles.css'
+
+class SlideComponent extends Component {
+  constructor(props) {
+    super(props)
+    this.imgRef = React.createRef()
+    this.handleResize = debounce(() => {
+      setTimeout(() => {
```

Comment:
Why the setTimeout?

Reply:
The fit doesn't work properly on resize if I don't put the setTimeout. The image doesn't get the proper resize

Reply:
This `setTimeout` is needed because of the animation duration of `Slider`. In the function `fit()` I take the current size of the `img` and the `parentNode`, if the browser is still applying the transition of width and height, I will get the wrong size and this will not work as it should. But instead of using `sliderTransitionDuration` I was using `resizeDebounce`, but I fixed it now.